da/da.go

```diff
@@ -60,7 +60,7 @@ type ResultRetrieveBlocks struct {
 // It also contains life-cycle methods.
 type DataAvailabilityLayerClient interface {
 	// Init is called once to allow DA client to read configuration and initialize resources.
-	Init(namespaceID types.NamespaceID, config []byte, kvStore store.KVStore, logger log.Logger) error
+	Init(namespaceID types.NamespaceID, config []byte, kvStore ds.Datastore, logger log.Logger) error
```

Comment:
@tzdybal are da clients suppose to the datastore in some ways? presently it is passed in the `Init` but never used.

Reply:
We just want to give them this possibility.